server/src/main/java/org/elasticsearch/action/bulk/FailureStoreDocumentConverter.java

```diff
@@ -24,6 +25,8 @@
  */
 public class FailureStoreDocumentConverter {
 
+    private static final List<String> INGEST_EXCEPTION_HEADERS = List.of("pipeline_origin", "processor_tag", "processor_type");
```

Comment:
Nit 1: since these headers are used 2+ times, should we introduce them as constants?

Nit 2: if we only use the contains method should we make it a set? Not so much for performance but more about hinting the way it's used.